### PR TITLE
Add Multi-projects/tasks  prometheus metric support

### PR DESCRIPTION
**Pull request by maxshuang**
close issue: #398

### CODE REVIEW

File: pkg/tenant/tenant.go [hunk 1]
```diff
@@ -1,11 +1,21 @@
...
+)
 
 // tenant const variables
 const (
 	FrameTenantID       = "dfe_root"
 	TestTenantID        = "dfe_test"
 	DefaultUserTenantID = "def_default_user"
 )
+
+// ProjectInfo is the tenant/project information which is consistent with DBaas
```
[the remark this answers is not in the document]
Comment (maxshuang) [reply pkg/tenant/tenant.go]: we don't need much work on tenant/project.
I think maintaining  both `tenantID` and `projectID` is fine.

File: pkg/promutil/implement_test.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
+package promutil
+
+import (
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/require"
+)
+
```
Comment (amyangfei) [inline pkg/promutil/implement_test.go:9]: Can we test every `NewXXX` methods of wrappingFactory

File: pkg/promutil/registry.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	cls, exists = r.collectorByWorker[workerID]
+	if !exists {
+		cls = make([]prometheus.Collector, 0)
+	}
+	cls = append(cls, c)
+	r.collectorByWorker[workerID] = cls
+}
+
+// Unregister unregisters all Collectors of the specified worker
+func (r *Registry) Unregister(workerID libModel.WorkerID) {
```
Comment (amyangfei) [inline pkg/promutil/registry.go:67]: We can add ad test to verify `Unregister` can work as expected

File: pkg/promutil/implement.go [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+	if prefix != "" {
+		opts.Namespace = prefix + "_" + opts.Namespace
+	}
+	cls := opts.ConstLabels
+	for name, value := range constLabels {
+		if _, exists := cls[name]; exists {
+			log.L().Panic("duplicate label name", zap.String("label", name))
+		}
+		cls[name] = value
+	}
```
Comment (amyangfei) [inline pkg/promutil/implement.go:86]: We can abstract this logic into a function and reuse it.

**Review by amyangfei — COMMENTED** [review]
rest lgtm

File: pkg/promutil/implement.go (hunk 4 above)
Comment (maxshuang) [reply pkg/promutil/implement.go]: solve it in last commit

**Review by amyangfei — APPROVED** [review]

**Pull request merged**